nav2_core/CMakeLists.txt

```diff
@@ -2,13 +2,46 @@ cmake_minimum_required(VERSION 3.5)
 project(nav2_core)
 
 find_package(ament_cmake REQUIRED)
-find_package(nav2_costmap_2d REQUIRED)
+find_package(nav2_common REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_lifecycle REQUIRED)
 find_package(std_msgs REQUIRED)
 find_package(geometry_msgs REQUIRED)
+find_package(nav2_costmap_2d REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(nav_msgs REQUIRED)
 find_package(tf2_ros REQUIRED)
 
+nav2_package()
+
+include_directories(
+  include
+)
+
+set(dependencies
```

Comment:
There's no code being built, the headers are just being copied into the include directory for others to pull in, the include_directories / set calls aren't necessary since we're just installing headers

Reply:
I think it makes sense to keep the set(*). We export those dependencies below, so that when somebody does find_package(nav2_core), it automatically adds in the nav2_costmap, pluginlib, std_msgs, etc dependencies. Otherwise, they'd need to explicitly add those in their package.